### PR TITLE
Use CMake as a build system

### PR DESCRIPTION
**Pull request by broadwaylamb**
This is my first PR in the framework of my term project, which goal is implementing the [LLVM](http://llvm.org/) backend for RuC. I've decided to start with a little cleanup to make the future work easier. In the subsequent pull requests I intend to do some refactoring, first touching the command line interface of the compiler, which is needed if we want to let the users choose which backend they want to leverage.

As a first step, in this PR we adopt CMake as a build system. [CMake](https://cmake.org) is used to control the compilation process using simple platform and compiler independent configuration files, and generate native makefiles and workspaces that can be used in the compiler environment of our choice.

The file `test.sh` is removed, and native CMake testing is used instead. It produces slightly more readable console output.

I've also modified the CI configurations so that the tests are now run on macOS (with Clang compiler) as well as Linux (with gcc).

You'll notice that the CI build is broken. That's OK — it's because not all the tests succeed, and they've been failing for a while, but the CI build was marked as "passing" because of the error in the `test.sh` file, namely this line:
https://github.com/andrey-terekhov/RuC/blob/d31e1d807fc2f5b7c34cc8a913147ec62c2af82e/test.sh#L56-L58
It literally says "IF the number of passed tests is not zero, exit with zero". Which is probably not what we want to do here.

This is fixed in this PR, the CI build is made ~great~ red again.

Some files are renamed — they contained some nasty characters like whitespaces or semicolons, CMake was unhappy.

### Motivation for using CMake
It's widely used in many C and C++ projects (including LLVM) as a better and more universal alternative to Makefiles and autotools. CMake provides with some level of independence of the OS and the compiler. It's also capable of generating a project for IDE of your choice, which is very convenient, since we can now debug the compiler faster.

CMake comes with a solution for testing called CTest, which is also leveraged in this pull request.
Later I'm planning to improve the testing infrastructure by adopting [lit](https://llvm.org/docs/CommandGuide/lit.html), the LLVM solution for integration testing the compiler. But for now let's stick with the current approach.

I've added some instructions to README, so that anyone even without previous experience with CMake can get our compiler up and running in just a couple of minutes.

I hope this PR won't get stuck like #1, since I really really believe that adopting CMake will make things better.

@andrey-terekhov @MikhailTerekhov can you please review and merge this?

### CODE REVIEW

**Comment by Victor-Y-Fadeev** [comment]
Well, we are interested in using CMake. But we have some remarks.

Firstly, don’t hardcode source files. Unfortunately, [Andrey N. Terekhov](https://github.com/andrey-terekhov) won’t follow its relevance. It is better using search by mask, as in tests.

Renaming test files isn’t a good idea. Because nobody won't maintain their correctness. A better solution would be automatically renaming before building. And ignoring incorrect names when testing.

About error in test.sh, it's not an error. The `./tests` folder contains files with deliberate errors for demonstration. These files aren't marked in any way and them always compilated with errors. Consequently, the test works as it is. To solve this problem, contact with [Andrey N. Terekhov](https://github.com/andrey-terekhov).

P.s. As a option, you can sort tests by some rules before building. In addition, it would be nice to add the execution of tests, which don't request console input. We are only compiling tests now. 

About Travis CI log, make it more readable. Remove `/Users/travis/build/andrey-terekhov/RuC` from each file’s path, it's clogging the space. Because sometimes we are printing this log. If it possible, colorize test's result, like in the [last master update](https://travis-ci.org/andrey-terekhov/RuC/builds/477028735).

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.0.0)
+
+project(RuC)
+
+set(CMAKE_C_STANDARD 11)
+
+if(ENABLE_SANITIZERS)
+  set(ENABLE_SANITIZERS "-g -fsanitize=address,undefined -fno-omit-frame-pointer")
```
Comment (Victor-Y-Fadeev) [inline CMakeLists.txt]: As for CMakeLists.txt, use a single style, like 4 spaces here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# Put the keywords.txt file near the driver
+file(GENERATE OUTPUT $<TARGET_FILE_DIR:${EXECUTABLE_NAME}>/keywords.txt
+              INPUT ${PROJECT_SOURCE_DIR}/keywords.txt)
+
+enable_testing()
+
+set(TEST_TIMEOUT 1)
+
+file(GLOB TEST_INPUTS "${TEST_INPUTS_DIR}/*.c"
```
Comment (Victor-Y-Fadeev) [inline CMakeLists.txt]: And no quotes here, as in the rest of file.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,60 @@
-RuC [![Build Status](https://travis-ci.org/andrey-terekhov/RuC.svg?branch=master)](https://travis-ci.org/andrey-terekhov/RuC)
-===
+# RuC
+[![Build Status](https://travis-ci.org/andrey-terekhov/RuC.svg?branch=master)](https://travis-ci.org/andrey-terekhov/RuC)
```
Comment (Victor-Y-Fadeev) [inline README.md]: Also not transfer Travis CI label to a new line, near the name it looks better.

**Comment by broadwaylamb** [comment]
@Victor-Y-Fadeev 

> About error in test.sh, it's not an error. The ./tests folder contains files with deliberate errors for demonstration. These files aren't marked in any way and them always compilated with errors. Consequently, the test works as it is. To solve this problem, contact with Andrey N. Terekhov.

Now I see. But does that mean that in order to detect a regression one would need to look through the entire list of tests in the console output and remember which of them are **expected** to fail and which are not?

**Comment by MikhailTerekhov** [comment]
> Alternatively, a more rough and simple way is to come to some sort of test naming convention, e. g. naming the files that are expected to produce compilation errors as `<test_name>_XFAIL.c`, and then use search by mask for such files. Or just put them in different directories.

There is no problem for authors to separate all tests into working and failing, if some way of sorting is provided. Moving failing tests into separate directory is OK.

**Pull request merged**